src/LaunchDarkly.EventSource/EventSource.cs

```diff
@@ -114,14 +125,22 @@ public EventSource(Configuration configuration)
         /// <exception cref="InvalidOperationException">The method was called after the connection <see cref="ReadyState"/> was Open or Connecting.</exception>
         public async Task StartAsync()
         {
-            var cancellationToken = _pendingRequest.Token;
-            while (!cancellationToken.IsCancellationRequested)
+            while (ReadyState != ReadyState.Shutdown)
             {
                 await MaybeWaitWithBackOff();
                 try
                 {
-                    await ConnectToEventSourceAsync(cancellationToken);
-                    _backOff.ResetReconnectAttemptCount();
```

Comment:
I've moved this call to `ResetReconnectAttemptCount` to [here](https://github.com/launchdarkly/dotnet-eventsource/pull/36/files#diff-896d0f336a16ef11d2f9c3bce9244546R215), to make it behave the way I think we always intended it to: it should reset the backoff count if it has successfully connected to the server. Putting it on line 124 was wrong because `await ConnectToEventSourceAsync` will only return when the connection has been closed.